submissions/chris-voitova/frogger-game/app.js

```diff
@@ -0,0 +1,138 @@
+const field = {
+  width: 505,
+  height: 606,
+  edges: {
+    x: 400,
+    y: 450,
+  },
+  waterPosition: 50,
+};
+
+// Enemies our player must avoid
+const Enemy = function (x, y, speed, width = 101, height = 61) {
+  // Variables applied to each of our instances go here,
+  // we've provided one for you to get started
+  this.x = x;
+  this.y = y;
+  this.speed = speed;
+  this.width = width;
+  this.height = height;
+
+  // The image/sprite for our enemies, this uses
+  // a helper we've provided to easily load images
+  this.sprite = "images/enemy-bug.png";
+};
+
+// Update the enemy's position, required method for game
+// Parameter: dt, a time delta between ticks
+Enemy.prototype.update = function (dt) {
+  // You should multiply any movement by the dt parameter
+  // which will ensure the game runs at the same speed for
+  // all computers.
+  this.x += this.speed * dt;
+  if (this.x > field.width) {
+    this.x = 0;
+  }
+};
+
+// Draw the enemy on the screen, required method for game
+Enemy.prototype.render = function () {
+  ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+// Now write your own player class
+// This class requires an update(), render() and
+// a handleInput() method.
+
+const Player = function (x, y, step = 50, width = 101, height = 101) {
```

Comment:
Same here

Reply:
also fixed

Reply:
The following is not a code review and not a requirement to fix anything. Consider it a source of ideas you may consider to use in the future.

You might have grouped constants like e.g.
```
// related constants in a common declaration
const playerWidth = 101,
  playerHeight = 101,
  playerFieldStep = 50;

// ... or related constants grouped in an object literal; 
// this also makes comment like "// player" unnecessary as the const name is pretty descriptive per se
const playerProps = {
  step: 50,
  width: 101,
  height: 101
};
```

Why grouping? This way you lessen the chances that someone else would insert an unrelated `const` declaration between related constants.

The above is true for `let` declarations as well.

You can also use variables and constants as default values in function declarations.
E.g.
```
const Player = function(x, y, width = playerProps.width) { 
// ...
```

P.S. I didn't check other changes.

Reply:
Thanks a lot! I understood your ideas, and thought about them. Finally, I reorganised my code for a better experience.